src/linear_map.rs

```diff
@@ -557,4 +571,17 @@ mod test {
 
         assert_eq!(Droppable::count(), 0);
     }
+
+    #[test]
+    fn into_iter() {
+        let mut src: LinearMap<_, _, 4> = LinearMap::new();
+        src.insert("k1", "v1").unwrap();
+        src.insert("k2", "v2").unwrap();
+        src.insert("k3", "v3").unwrap();
+        src.insert("k4", "v4").unwrap();
+        let clone = src.clone();
+        for (k, v) in clone.into_iter() {
+            assert_eq!(v, *src.get(k).unwrap());
```

Comment:
```suggestion
            assert_eq!(v, src.remove(k).unwrap());
```

Reply:
Thank you for the suggestion. Applied in [a5f7a67](https://github.com/rust-embedded/heapless/pull/474/commits/a5f7a67f60f627af0e85000dc2f76f1dfdeb58f3)